### ISSUE TITLE
Aggregation.__call__ calls nn.Module.__call__ using (x, index, ptr, dim_size, dim, **kwargs) instead of (inputs=x, index=index, ptr=ptr, dim_size=dim_size, dim=dim, **kwargs) 

### ISSUE THREAD

**Issue by bio-mlhui** [pyg-team/pytorch_geometric#7979]
### 🛠 Proposed Refactor

if somebody wants to inheric Aggregation base class, AND wants to add some conditional data used in aggregation, he must re-define forward as (x: Tensor, index: Tensor=None, ptr=None, dim_size: int=None, dim: int = -2, **condition: Tensor=None**), since the internal call of Aggregation to nn.Module uses fixed positional args. I think it is more elegant to define forward as (x, condition, index=None, ptr=None, dim_size=None, dim=None)

### Suggest a potential alternative/fix

As suggested

**Comment by rusty1s** [comment]
I agree. Will be fixed in https://github.com/pyg-team/pytorch_geometric/pull/7990.
